src/material/dialog/dialog.md

```diff
@@ -173,6 +173,18 @@ be configured by setting the `cdkFocusInitial` attribute on another focusable el
 Tabbing through the elements of the dialog will keep focus inside of the dialog element,
 wrapping back to the first tabbable element when reaching the end of the tab sequence.
 
+### Focus Restoration
+After closing a dialog the focus is restored to the last focused element before the dialog was opened. 
+However this will not work if the dialog is opened from a transient element (eg. a menu item). If you 
+would like to restore the focus to a different element (eg. menu trigger), you can disable `restoreFocus` 
+in `MatDialogConfig` either by setting it into the `open` method or passing it in as the value of 
+`MAT_DIALOG_DEFAULT_OPTIONS` in the providers.
```

Comment:
I wouldn't recommend `MAT_DIALOG_DEFAULT_OPTIONS` since this is more of an edge case and that would change the configuration more broadly.